### PR TITLE
Feature/apibackend form hints

### LINKED ISSUE #427: Add or edit help text for the fields of Add API Backend form
The Add API Backend form is poorly documented. Add descriptive notes to all fields on the form, so that users can understand their purpose.

Definition of done
---
* [x] Create the [Add API Backend Form help text document](https://github.com/apinf/api-umbrella-dashboard/wiki/Add-API-Backend-form-help-text) to create draft copy for help text
* [ ] Update [Add API Backend Form help text document](https://github.com/apinf/api-umbrella-dashboard/wiki/Add-API-Backend-form-help-text) with relevant help text for each field
* [ ] Update the helps texts on the HTML form.

### PR DESCRIPTION
**Pull request by elnzv**
Closes #427, #431

### CODE REVIEW

**Comment by brylie** [comment]
Status update?

**Comment by elnzv** [comment]
Fixed conflicts.
Needed a pair programming session with @apinf/developers to place hints' text.

**Comment by brylie** [comment]
One last change is to add `"'InlineHelp": false` to the `.jshintrc` file under the `// Apinf packages` section.

**Comment by elnzv** [comment]
Closes #427 
@apinf/developers, please review amd merge.

**Comment by elnzv** [comment]
New hints' text added.
WIP removed and PR is ready to be reviewed and merged.
@apinf/developers, please review it ASAP.

File: client/views/api_backends/form_hints/form_hints.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  },
+  'set_headers': {
+    message: "If your API requires some specific headers to be set, they should be added here",
+    options: {
+      placement: 'left'
+    }
+  },
+  'http_basic_auth': {
+    message: "Does your api need a username and password for access? If yes, the must be specified here.",
```
Comment (frenchbread) [inline client/views/api_backends/form_hints/form_hints.js]: typo in second sentence. I think it should be **they** rather than **the**.

**Comment by frenchbread** [comment]
@elnzv looks good! But found some issues:

* Can this design issue be fixed to be like inline or with paddings from top and right?

   screen shot 2015-10-20 at 18 27 31

* Typo in `client/views/api_backends/form_hints/form_hints.js` 69 line
* Also found out that some fields do not have inline help support. Fields like `required_roles`, `rate_limit_mode` and others .. Is that a bug or a feature?

**Comment by bajiat** [comment]
@elnzv Is there any easy way for me to see the texts in the actual form?

**Comment by elnzv** [comment]
@bajiat, I have just copied text from Aseem's letter.
Here are the list of them - https://github.com/apinf/api-umbrella-dashboard/blob/feature/apibackend-form-hints/client/views/api_backends/form_hints/form_hints.js

**Comment by elnzv** [comment]
@frenchbread, fixed typo, added padding, yes, for now these are only available hints, further on we can add new or change existing ones.

**Comment by frenchbread** [comment]
@elnzv Thanks for fixes. Merging..

**Pull request merged**